src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexProvider.cs

```diff
@@ -32,80 +23,37 @@ internal class FunctionIndexProvider : IFunctionIndexProvider
             IFunctionExecutor executor,
             IExtensionRegistry extensions,
             SingletonManager singletonManager,
-            TraceWriter trace)
+            TraceWriter trace,
+            IWebJobsExceptionHandler handler)
         {
             if (typeLocator == null)
             {
-                throw new ArgumentNullException("typeLocator");
-            }
-
-            if (triggerBindingProvider == null)
-            {
-                throw new ArgumentNullException("triggerBindingProvider");
-            }
-
-            if (bindingProvider == null)
-            {
-                throw new ArgumentNullException("bindingProvider");
-            }
-
-            if (activator == null)
```

Comment:
FunctionIndexer had all null checks for its parameters so I decided that it's fine to omit them here